picard/ui/options/renaming.py

```diff
@@ -152,24 +150,29 @@ def update_selector_from_editor(self):
         """
         self.naming_scripts = self.script_editor_dialog.naming_scripts
         self.selected_naming_script_id = self.script_editor_dialog.selected_script_id
-        self.update_selector_from_settings()
+        populate_script_selection_combo_box(self.naming_scripts, self.selected_naming_script_id, self.ui.naming_script_selector)
+        self.display_examples()
 
     def update_selector_from_settings(self):
         """Update the script selector combo box from the settings.
         """
         populate_script_selection_combo_box(self.naming_scripts, self.selected_naming_script_id, self.ui.naming_script_selector)
+        self.update_selector_in_editor()
 
-    def update_selector_in_editor(self):
+    def update_selector_in_editor(self, skip_check=False):
         """Update the selection in the script editor page to match local selection.
+
+        Args:
+            skip_check (bool): Skip the check for unsaved edits.  Defaults to False.
         """
-        script_item = self.ui.naming_script_selector.itemData(self.ui.naming_script_selector.currentIndex())
         if self.script_editor_dialog:
-            self.script_editor_dialog.set_selected_script_index(self.ui.naming_script_selector.currentIndex(), send_signal=False)
-        self.script_text = script_item["script"]
-        self.selected_naming_script_id = script_item["id"]
-        override = {'file_naming_format': self.script_text}
-        self.examples.update_examples(override)
-        self.update_examples_from_local()
+            self.script_editor_dialog.set_selected_script_index(self.ui.naming_script_selector.currentIndex(), skip_check=skip_check)
+        else:
+            script_item = self.ui.naming_script_selector.itemData(self.ui.naming_script_selector.currentIndex())
```

Comment:
Store `self.ui.naming_script_selector.currentIndex()` in a variable and shorten lines above